src/Dali/RedSharp.Dali.Controls/Converters/ColourBrightnessConverter.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using System.Windows.Data;
+using System.Windows.Media;
+
+namespace RedSharp.Dali.Controls.Converters
+{
+    /// <summary>
+    /// Converter that calculates brighter and darker colours based on given one.
+    /// </summary>
+    class ColourBrightnessConverter : IValueConverter
+    {
+        /// <summary>
+        /// Performs actual calculation of colour.
+        /// </summary>
+        /// <param name="colour">Base colour.</param>
+        /// <param name="coef">Coeficient of brighness. Any number more or equat to 0. Values
+        /// less than 1 cause darker colour, 1 the same one, larger then 1 - brighter.</param>
+        /// <returns>Darker or brighter brush.</returns>
+        private SolidColorBrush ModifyColour(Color colour, double coef)
+        {
+            byte R, G, B;
+
+            R = Math.Min((byte)255, (byte)(colour.R * coef));
+            G = Math.Min((byte)255, (byte)(colour.G * coef));
+            B = Math.Min((byte)255, (byte)(colour.B * coef));
```

Comment:
Bad code, you must do Math.Min() and then do type conversion, in another cases you will have wrong result.

![image](https://user-images.githubusercontent.com/43117099/71783520-8350f980-2ff0-11ea-9a5b-bcc5bc9ad5f7.png)

Reply:
Also, can you make this methods as internal? I think cases like this can be covered by unit tests in future.

Reply:
One more "also", ModifyColour - it's a very very general name for those methods(I saw several like this one next below). Will it make it difficult to concretize names of those methods?